### PR TITLE
Fix/allow registry to be zero address

### PR DESCRIPTION
**Pull request by VGabriel45**

### CODE REVIEW

File: contracts/factory/K1ValidatorFactory.sol [hunk 1]
```diff
@@ -63,11 +63,7 @@ contract K1ValidatorFactory is Stakeable {
         IERC7484 registry
     ) Stakeable(factoryOwner) {
         require(
-            !(implementation == address(0) ||
-                k1Validator == address(0) ||
-                address(bootstrapper) == address(0) ||
-                address(registry) == address(0) ||
-                factoryOwner == address(0)),
+            !(implementation == address(0) || k1Validator == address(0) || address(bootstrapper) == address(0)),
```
Comment (livingrockrises) [inline contracts/factory/K1ValidatorFactory.sol:66]: why this is skipped too?
factoryOwner == address(0)

File: test/foundry/unit/concrete/factory/TestK1ValidatorFactory_Deployments.t.sol [hunk 2]
```diff
@@ -50,6 +50,29 @@ contract TestK1ValidatorFactory_Deployments is NexusTest_Base {
         assertTrue(isContract(address(factory)), "Factory should be a contract");
     }
 
+    /// @notice Tests that the constructor can take a zero address for the registry.
+    function test_ConstructorInitializesWithRegistryAddressZero() public {
```
Comment (livingrockrises) [inline test/foundry/unit/concrete/factory/TestK1ValidatorFactory_Deployments.t.sol:54]: we would also have 7484 tests somewhere with some mock registry.
we can set it to zero and see if everything is going fine. 
cases I mentioned on slack thread once.

File: contracts/factory/K1ValidatorFactory.sol (hunk 1 above)
Comment (VGabriel45) [reply contracts/factory/K1ValidatorFactory.sol:66]: Weird, it was not removed on my local version

File: test/foundry/unit/concrete/factory/TestK1ValidatorFactory_Deployments.t.sol (hunk 2 above)
Comment (VGabriel45) [reply test/foundry/unit/concrete/factory/TestK1ValidatorFactory_Deployments.t.sol:54]: Not sure which tests you mean

**Review by livingrockrises — APPROVED** [review]
looks good to go. check comment about test cases. If done in different branch for coverage improvement then I can merge this

**Pull request merged**